src/components/core/Table.tsx

```diff
@@ -88,65 +89,79 @@ export class Table extends React.Component<ITableProps, ITableState> {
+        const { dataSource, resultSpec } = this.props;
+        this.initDataLoading(dataSource, resultSpec);
     }
 
     public componentWillReceiveProps(nextProps: ITableProps) {
-        const sortingPrev = get<IVisualizationProperties, ISorting>(this.props.visualizationProperties, 'sorting');
-        const sortingNext = get<IVisualizationProperties, ISorting>(nextProps.visualizationProperties, 'sorting');
+        const sortItemsPrev = get<IVisualizationProperties, AFM.SortItem[]>(
+            this.props.visualizationProperties, 'sortItems'
+        );
+        const sortItemsNext = get<IVisualizationProperties, AFM.SortItem[]>(
+            this.props.visualizationProperties, 'sortItems'
+        );
         // next sorting needs to be different from previous and also
         // than actual inner sorting to get rid of duplicate execution
         // This handles only UNDO sorting change
-        const sortingChanged = !isEqual(sortingPrev, sortingNext) && !isEqual(this.state.sorting, sortingNext);
+        const sortingChanged = !isEqual(sortItemsPrev, sortItemsNext) && !isEqual(this.state.sortItems, sortItemsNext);
 
         if (!DataSourceUtils.dataSourcesMatch(this.props.dataSource, nextProps.dataSource) || sortingChanged) {
-            if (this.dataCancellable) {
-                this.dataCancellable.cancel();
-            }
-
-            const sorting: ISorting = sortingChanged ? sortingNext : this.state.sorting;
+            const sortItems = sortingChanged ? sortItemsNext : this.state.sortItems;
 
-            const { metadataSource, dataSource, transformation } = nextProps;
-            this.initDataLoading(dataSource, metadataSource, transformation, sorting);
+            const { dataSource, resultSpec } = nextProps;
+            this.initDataLoading(dataSource, resultSpec, sortItems);
         }
     }
 
     public componentWillUnmount() {
-        if (this.dataCancellable) {
-            this.dataCancellable.cancel();
-        }
+        this.subscription.unsubscribe();
+        this.subject.unsubscribe();
         this.onLoadingChanged = noop;
         this.onError = noop;
     }
 
-    public onSortChange(change: ISortingChange) {
-        const sorting = getSorting(change, get<ISorting, Transformation.ISort>(this.state.sorting, 'sorting'));
-        const sortingInfo = {
-            sorting, change
-        };
+    public onSortChange(sortItems: AFM.SortItem[]) {
         this.setState({
-            sorting: sortingInfo
+            sortItems
         });
 
-        const { metadataSource, dataSource, transformation } = this.props;
+        const { dataSource, resultSpec } = this.props;
         this.props.pushData({
             properties: {
-                sorting: sortingInfo
+                sortItems
             }
         });
-        this.initDataLoading(dataSource, metadataSource, transformation, sortingInfo);
+        this.initDataLoading(dataSource, resultSpec);
```

Comment:
Don't you want to use new sortItems for dataLoading?

Reply:
good catch :)